tool/tsh/config.go

```diff
@@ -19,44 +19,59 @@ package main
 import (
 	"fmt"
 	"net"
+	"os"
+	"os/exec"
+	"runtime"
 	"strings"
 	"text/template"
 
+	"github.com/gravitational/teleport/api/constants"
 	"github.com/gravitational/teleport/api/profile"
 	"github.com/gravitational/teleport/api/utils/keypaths"
 	"github.com/gravitational/trace"
 )
 
 const sshConfigTemplate = `
-# Common flags for all {{ .clusterName }} hosts
-Host *.{{ .clusterName }} {{ .proxyHost }}
-    UserKnownHostsFile "{{ .knownHostsPath }}"
-
-# Flags for all {{ .clusterName }} hosts except the proxy
-Host *.{{ .clusterName }} !{{ .proxyHost }}
+# Common flags for all {{ .ClusterName }} hosts
+Host *.{{ .ClusterName }} {{ .ProxyHost }}
+    UserKnownHostsFile "{{ .KnownHostsPath }}"
+    IdentityFile "{{ .IdentityFilePath }}"
+    CertificateFile "{{ .CertificateFilePath }}"
+
+# Flags for all {{ .ClusterName }} hosts except the proxy
+Host *.{{ .ClusterName }} !{{ .ProxyHost }}
     Port 3022
-    {{- if .leaf }}
-    ProxyCommand ssh -p {{ .proxyPort }} {{ .proxyHost }} -s proxy:$(echo %h | cut -d '.' -f 1):%p@{{ .clusterName }}	
-    {{- else }}
-    ProxyCommand ssh -p {{ .proxyPort }} {{ .proxyHost }} -s proxy:$(echo %h | cut -d '.' -f 1):%p
-    {{- end }}
+    ProxyCommand {{ .TSHPath }} config-proxy --login="%r" --proxy={{ .ProxyHost }}:{{ .ProxyPort }} %h:%p "{{ .ClusterName }}"
 `
 
+type hostConfigParameters struct {
+	ClusterName         string
+	KnownHostsPath      string
+	IdentityFilePath    string
+	CertificateFilePath string
+	ProxyHost           string
+	ProxyPort           string
+	TSHPath             string
+}
+
+// getSSHPath returns a sane default `ssh` path for the current platform.
+func getDefaultSSHPath() string {
```

Comment:
I've switched over to using `LookPath` on all platforms, after some testing the main other SSH binary we're likely to see on Windows (from Git for Windows) works perfectly fine. This should make Git for Windows a viable client on older Windows versions since #7790 only supports Win10.